### PR TITLE
Creates script command for sql-cache-tool

### LINKED ISSUE #24635: Add script command to sql-cache tool
### Is your feature request related to a problem? Please describe.
I am trying to create the tables for distributed sql cache via the dotnet tool "sql-cache".
This works well, as long a I am developing and have full database access, but it's not working for scenarios, where I lack of access rights.

### Describe the solution you'd like
Add `dotnet sql-cache script`, so the neccessary commands are written to the stdout and can easily be copied to an textfile.

### PR DESCRIPTION
**Pull request by joncloud**
- [X] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [X] You've included unit or integration tests for your change, where applicable.
- [X] You've included inline docs for your change, where applicable.
- [X] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

**PR Title**
Creates a new command for sql-cache-tool, which outputs the SQL Script needed.

**PR Description**
* Adds new `script` command to create SQL Script
* Includes the ability to output directly to a file
* Includes the ability to make the output script idempotent
* Minor fix to `create` command to produce the command's help on validation error

Addresses #24635

### CODE REVIEW

**Comment by mkArtakMSFT** [comment]
@ajcvickers does you team own this tool?

**Comment by ajcvickers** [comment]
@mkArtakMSFT Nope.

**Comment by mkArtakMSFT** [comment]
@BrennanConroy who owns this? can you please assign someone to review this? Thanks!

**Comment by BrennanConroy** [comment]
Command line tools has been under your team https://github.com/dotnet/aspnetcore-internal/blob/master/docs/engineering/SMEs.md

Should we bring this up at sync today?

**Comment by Pilchie** [comment]
@davidfowl - can you help take a look at this?

**Comment by davidfowl** [comment]
The overall change looks reasonable. I'd need to look at the SQL output since we have no tests for this stuff it seems.

**Comment by joncloud** [comment]
I created a [quick gist](https://gist.github.com/joncloud/0415994d0b78f97d31b21592389ea108) of the output, which includes

* SqlCache.idempotent.sql (run with args `script dbo SqlCache --idempotent`)
* SqlCache.sql (run with args `script dbo SqlCache`)

If you would like to see some tests, I'd be happy to create a new project to build out some test automation. It'd be great to get a little bit of guidance for how you'd like to see the integration test written. I did find `src/Caching/SqlServer/test/SqlServerCacheWithDatabaseTest.cs`, which is testing the actual schema that is created by this program.

**Comment by davidfowl** [comment]
@ajcvickers could you help review this? Needs somebody with SQL expertise to eyeball the query. It LGTM

**Comment by ajcvickers** [comment]
@ErikEJ would you mind taking a look as a SQL expert?

**Review by ErikEJ — COMMENTED** [review]
LGTM

File: src/Tools/dotnet-sql-cache/src/Program.cs [hunk 1]
```diff
@@ -82,6 +86,51 @@ public int Run(string[] args)
...
+
+                    var tableNameArg = command.Argument(
+                        "[tableName]", "Name of the table to be created.");
+
+                    var outputOption = command.Option(
+                        "-o|--output",
+                        "The file to write the result to.",
+                        CommandOptionType.SingleValue);
+
+                    var idempontentOption = command.Option(
```
[the remark this answers is not in the document]
Comment (joncloud) [reply src/Tools/dotnet-sql-cache/src/Program.cs:104]: No particular reason. The original request did not ask for the idempotent option, but I modeled the options based on how `dotnet ef migrations script` worked. If it makes more sense to always output an idempotent script, I can rework this. Let me know.

**Review by davidfowl — APPROVED** [review]

**Comment by Pilchie** [comment]
@davidfowl want do the honors and merge?

**Pull request merged**